ui/app/AppLayouts/Communities/panels/EditSettingsPanel.qml

```diff
@@ -62,12 +71,71 @@ StatusScrollView {
             Layout.fillWidth: true
         }
 
+        StatusModalDivider {
+            Layout.fillWidth: true
+            Layout.topMargin: -baseLayout.spacing
+            Layout.bottomMargin: 2
+            visible: root.communityShardingEnabled
+        }
+
+        RowLayout {
+            spacing: Style.current.halfPadding
+            visible: root.communityShardingEnabled
+
+            readonly property bool shardingActive: root.communityShardIndex !== -1
+
+            StatusBaseText {
+                Layout.fillWidth: true
+                text: qsTr("Community sharding")
+            }
+            Item { Layout.fillWidth: true }
+            StatusBaseText {
+                color: Theme.palette.baseColor1
+                visible: parent.shardingActive
+                text: qsTr("Active: on shard #%1").arg(root.communityShardIndex)
+            }
+            StatusButton {
+                size: StatusBaseButton.Size.Small
+                text: parent.shardingActive ? qsTr("Manage") : qsTr("Make %1 a sharded community").arg(root.name)
+                onClicked: parent.shardingActive ? Global.openPopup(manageShardingPopupCmp) : Global.openPopup(enableShardingPopupCmp)
+            }
+        }
+
         Item {
             // settingsDirtyToastMessage placeholder
             visible: root.bottomReservedSpaceActive
             implicitWidth: root.bottomReservedSpace.width
             implicitHeight: root.bottomReservedSpace.height
         }
+
+        Component {
+            id: enableShardingPopupCmp
+            EnableShardingPopup {
+                destroyOnClose: true
+                communityName: root.name
+                publicKey: root.communityId
+                shardingInProgress: false // TODO community sharding backend: set to "true" when generating the pubSub topic, or migrating
+                onEnableSharding: {
+                    console.warn("TODO: enable community sharding for shardIndex:", shardIndex) // TODO community sharding backend
+                    root.communityShardIndex = shardIndex
```

Comment:
WDYT of replacing this with signals? We'll need them anyway to integrate the backend. Probably we need to forward the signal all the way to `CommunitySettingsView`.

Reply:
Hmm, I'd leave that for later; could be a signal, or some store function

Reply:
There's no store access for these views. This is why I was thinking the signal could be safely implemented in this PR. We probably won't want any store access to be added here later on 😄
Adding it later is fine as well.